Tiltfile

```diff
@@ -61,11 +62,89 @@ def deploy_capi():
             if kb_extra_args:
                 patch_args_with_extra_args("capi-kubeadm-bootstrap-system", "capi-kubeadm-bootstrap-controller-manager", kb_extra_args)
 
+## This should have the same versions as the Dockerfile
+if settings.get("local_mode"):
+    tilt_dockerfile_header_cso = """
+    FROM ghcr.io/sovereigncloudstack/cso:v0.1.0-alpha.2 as builder
+
+    FROM docker.io/library/alpine:3.18.0 as tilt
+    WORKDIR /
+    COPY --from=builder /usr/local/bin/helm /usr/local/bin/helm
+    COPY --from=builder /manager /manager
+    COPY .release/ /tmp/downloads/cluster-stacks/
+    """
+else:
+    tilt_dockerfile_header_cso = """
+    FROM ghcr.io/sovereigncloudstack/cso:v0.1.0-alpha.2 as builder
+
+    FROM docker.io/library/alpine:3.18.0 as tilt
+    WORKDIR /
+    COPY --from=builder /usr/local/bin/helm /usr/local/bin/helm
+    COPY --from=builder /manager /manager
+    """
+    
+
 def deploy_cso():
-    version = settings.get("cso_version")
-    cso_uri = "https://github.com/sovereignCloudStack/cluster-stack-operator/releases/download/{}/cso-infrastructure-components.yaml".format(version)
-    cmd = "curl -sSL {} | {} | kubectl apply -f -".format(cso_uri, envsubst_cmd)
-    local(cmd, quiet = True)
+    # yaml = str(kustomizesub("./hack/observability")) # build an observable kind deployment by default
+    yaml_cso = './cso-components.yaml'
+    local_resource(
+        name = "cso-components",
+        cmd = ["sh", "-ec", sed_cmd, yaml_cso, "|", envsubst_cmd],
+        labels = ["CSO"],
+    )
+
+    entrypoint = ["/manager"]
+    extra_args = settings.get("extra_args")
+    if extra_args:
+        entrypoint.extend(extra_args)
+
+    # Set up an image build for the provider. The live update configuration syncs the output from the local_resource
+    # build into the container.
+    if settings.get("local_mode"):
+        docker_build_with_restart(
+            ref = "ghcr.io/sovereigncloudstack/cso-staging",
```

Comment:
Since we plan on using `cso-components.yaml` directly from GitHub. I think using `ghcr.io/sovereigncloudstack/cso:v0.1.0-alpha.2` this sounds better. I'll update accordingly.